hadoop-tools/hadoop-azure/src/test/resources/abfs-testrun-configs.xml.template

```diff
@@ -0,0 +1,156 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+<configuration>
+
+  <!--====================== IMPORTANT!! ========================-->
+  <!--
+      1. PLEASE PROVIDE VALUES FOR THE CONFIGS IN THIS FILE AND
+         RENAME THE SAME TO "abfs-testrun-configs.xml"
+      2. UPDATE runtests.sh
+         A.  "fs.azure.abfs.account.name" AND "fs.azure.test.namespace.enabled"
+             ARE MANDATORY WITH EVERY SCENARIO. AUTHTYPE BY DEFAULT WILL BE
+             SHAREDKEY AS CONFIGURED IN THIS TEMPLATE FILE.
+         B.  PLEASE ADD MORE SCENARIOS IF THE CODE CHANGE REQUIRES TESTING WITH
+             DIFFERENT VARIANTS OF CONFIGS.
+      3. THE SCRIPT REQUIRES THE FOLLOWING UTILITIES xmlstarlet AND pcregrep
```

Comment:
??

Reply:
xmlstarlet is used to manipulate the xml file. We run various combinations of configs by manipulating the abfs-testrun-config.xml file.
pcregrep is used to extract multi line patterns matching a regex. We use this to extract the result lines for each combination of configs and output that to the consolidated results log file (The content of the same has to be updated in the PR).

Reply:
OK. Is it documented where to get these from? homebrew on the mac finds xmlstarlet, but not pcgrep

Reply:
It is **pcregrep**
https://stackoverflow.com/questions/44112312/how-do-i-install-pcre-with-homebrew